README.md

```diff
@@ -8,12 +8,47 @@ The goal of [Accessors.jl](https://github.com/JuliaObjects/Accessors.jl) is to m
 It is the successor of [Setfield.jl](https://github.com/jw3126/Setfield.jl).
 
 # Usage
-Updating immutable data was never easier:
+Say you have some immutable data structure, such as a `NamedTuple`:
 ```julia
-using Accessors
-@set obj.a.b.c = d
```

Comment:
Can you keep the original three lines as TLDR?

Reply:
That is possible, but I am hesitant about it. It still looks to me like it should update `obj`, so if I only read that, I find it misleading. But having the mutating version in a line below should alert the user to check out both, something like 
```
using Accessors
@set obj.a.b.c = d     # Non-mutating
@reset obj.a.b.c = d   # Mutating
```
would resolve that issue for me. Do you find that to be a good solution? I still feel like that splits the "tutorial" in 3 - the TLDR, the longer version below, and the actual docs. It feels like too much fracturing in documentation for a quite simple package to me... But if you much prefer it, it is fine by me.

Slightly separate but related issue; the name `reset` does not sit well with me. I intuitively expect it to change the fields changed by `@set` back to their initial value, resetting them. The current `@reset` does not in fact "reset" anything. I would much prefer `@set` and `@set!`. Is such an API change feasible? Or at least alias `@set!` to reset? (Or the other way around).

Reply:
If we agree that `@set!` is better than `@reset`, I suggest the following:
Define `@set!` to do what `@reset` currently does (non-breaking), bind `const @reset = @set!`, print deprecation warnings when `@reset` is used (because in the long term, we do not want 2 macros doing the same thing) for some months/a year, and then deprecate `@set!` in some future breaking release.

Reply:
>  I still feel like that splits the "tutorial" in 3 - the TLDR, the longer version below, and the actual docs.

Good point, we can drop the TLDR;

> If we agree that `@set!` is better than `@reset`

In Setfield.jl we have `@set!` instead of `@restet` and I consider it a bad name. To me it is important to distinguish between rebinding a variable and actual mutation. Mutation is a notorious source of complexity and bugs, while rebinding a variable is a very local change and does not cause problems in my experience.

Reply:
I see! Alright, so `@set!` is misleading, as nothing is actually mutated.

However, we still have a problem to fix IMO: `@reset` makes it sound like something is going back to some original state, i.e. being "reset". This is misleading, as the new state is previously unseen. How about the quite plain and accurate `@rebind`, with as many chars as `@reset`?

Reply:
I agree that `@reset` is not ideal. `@rebind` might be a bit better, but I don't find it to be better by a large enough margin to make a change.

Reply:
`@selfset` was another candidate name I considered but did not find good enough to justify the change.

Reply:
Reset is also similar enough to set that it add some consistency, so it is fint by me